### PR TITLE
Migration page for Dart and Flutter v7

### PR DESCRIPTION
**Pull request by marandaneto**
https://github.com/getsentry/sentry-dart/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+milestone%3A7.0.0

### CODE REVIEW

**Review by krystofwoldrich — APPROVED** [review]
LGTM 🚀

File: src/platforms/dart/migration.mdx [hunk 1]
```diff
@@ -4,6 +4,43 @@ sidebar_order: 1000
...
+  - `language`, using the `SentryCulture#locale` instead.
+  - `theme`, using the `SentryOperatingSystem#theme` instead.
+- Remove deprecated field from the `Contexts#dart_context` databag.
+  - `isolate`, using the `SentryThread#name` instead.
+- `SentryMeasurementUnit` has been redesigned to be strongly typed.
+  - `DurationSentryMeasurementUnit`
+  - `InformationSentryMeasurementUnit`
+  - `FractionSentryMeasurementUnit`
+  - `CustomSentryMeasurementUnit`
+  - `NoneSentryMeasurementUnit`
```
Comment (lizokm) [inline src/platforms/dart/migration.mdx:25]:
```suggestion
-  The following`SentryMeasurementUnit`s are now strongly typed:

  - `DurationSentryMeasurementUnit`
  - `InformationSentryMeasurementUnit`
  - `FractionSentryMeasurementUnit`
  - `CustomSentryMeasurementUnit`
  - `NoneSentryMeasurementUnit`
```

File: src/platforms/dart/migration.mdx [hunk 2]
```diff
@@ -4,6 +4,43 @@ sidebar_order: 1000
...
+- Remove deprecated fields from the `Scope` class.
+  - `user(SentryUser? user)`, using the `setUser(SentryUser? user)` instead.
+  - `attachements`, using the `attachments` instead.
```
Comment (lizokm) [inline src/platforms/dart/migration.mdx:28]:
```suggestion
- The following fields have been removed from the `Scope` class and replaced:

  - `user(SentryUser? user)` replaced with `setUser(SentryUser? user)`.
  - `attachements` replaced with `attachments`.
```

File: src/platforms/dart/migration.mdx [hunk 3]
```diff
@@ -4,6 +4,43 @@ sidebar_order: 1000
...
+- Classes or methods that used to take the bellow optional parameters were moved to the `SentryOptions` class.
+  - `captureFailedRequests`, using `SentryOptions#captureFailedRequests` instead.
+  - `sendDefaultPii`, using `SentryOptions#sendDefaultPii` instead.
+  - `maxRequestBodySize`, using `SentryOptions#maxRequestBodySize` instead.
+  - `networkTracing`, using `SentryOptions#tracesSampleRate` or `SentryOptions#tracesSampler` instead.
+  - `recordBreadcrumbs`, using `SentryOptions#recordHttpBreadcrumbs` instead.
```
Comment (lizokm) [inline src/platforms/dart/migration.mdx:34]:
```suggestion
- Classes or methods that used to take the bellow optional parameters, have been moved to the `SentryOptions` class:

  - `captureFailedRequests` replaced with `SentryOptions#captureFailedRequests` instead.
  - `sendDefaultPii` replaced with `SentryOptions#sendDefaultPii` instead.
  - `maxRequestBodySize` replaced with `SentryOptions#maxRequestBodySize` instead.
  - `networkTracing` replaced with `SentryOptions#tracesSampleRate` or `SentryOptions#tracesSampler` instead.
  - `recordBreadcrumbs` replaced with `SentryOptions#recordHttpBreadcrumbs` instead.
```

**Review by lizokm — COMMENTED** [review]
Hi! I'm not certain that I'm interpreting this correctly. Please let me know if I've changed the intended meaning or if you want to talk about any of the suggestions.

File: src/platforms/dart/migration.mdx [hunk 4]
```diff
@@ -4,6 +4,43 @@ sidebar_order: 1000
...
+
+- Sentry's Dart SDK version 7.0.0 and above requires Dart `2.17.0`.
+- Methods that used to take a `dynamic hint` optional parameter now takes `Hint? hint` instead.
+- Remove deprecated fields from the `SentryDevice` class.
+  - `screenResolution`, using the `screenHeightPixels` and `screenWidthPixels` instead.
+  - `timezone`, using the `SentryCulture#timezone` instead.
+  - `language`, using the `SentryCulture#locale` instead.
+  - `theme`, using the `SentryOperatingSystem#theme` instead.
+- Remove deprecated field from the `Contexts#dart_context` databag.
+  - `isolate`, using the `SentryThread#name` instead.
```
[the remark this answers is not in the document]
Comment (marandaneto) [reply src/platforms/dart/migration.mdx:20]: Nope, `isolate` is the name of the field, so I'd say its similar to the suggestions above, using the "replaced with" wording.

File: src/platforms/dart/migration.mdx [hunk 5]
```diff
@@ -4,6 +4,47 @@ sidebar_order: 1000
...
+- Sentry's Dart SDK version 7.0.0 and above requires Dart `2.17.0`.
+- Methods that used to take a `dynamic hint` optional parameter now take `Hint? hint` instead.
+- The following deprecated fields have been removed from the `SentryDevice` class and replaced:
+
+  - `screenResolution` replaced with `screenHeightPixels` and `screenWidthPixels`.
+  - `timezone` replaced with `SentryCulture#timezone`.
+  - `language` replaced with `SentryCulture#locale`.
+  - `theme` replaced with `SentryOperatingSystem#theme`.
+- Remove deprecated field from the `Contexts#dart_context` databag.
+  - `isolate`, using the `SentryThread#name` instead.
```
Comment (marandaneto) [inline src/platforms/dart/migration.mdx:20]:
```suggestion
- The following deprecated field has been removed from the `Contexts#dart_context` databag:
  - `isolate` replaced with `SentryThread#name`.
```

File: src/platforms/dart/migration.mdx (hunk 4 above)
[the remark this answers is not in the document]
Comment (marandaneto) [reply src/platforms/dart/migration.mdx]: [`35de412` (#6090)](https://github.com/getsentry/sentry-docs/pull/6090/commits/35de412136594524f3478a2d4824e229b1585e94)

File: src/platforms/flutter/migration.mdx [hunk 6]
```diff
@@ -3,6 +3,20 @@ title: Migration Guide
...
+
+API changes:
+
+- Sentry's Flutter SDK version 7.0.0 and above requires Flutter `3.0.0`.
+- The Sentry Cocoa SDK was upgraded to `8.0.0` which introduces breaking changes, see the [migration guide](/platforms/apple/migration/#migrating-from-7x-to-8x).
+- The following fields have been removed from the `SentryFlutterOptions` class and replaced:
+  - `enableAutoPerformanceTracking` replaced with `enableAutoPerformanceTracing`.
+  - `enableOutOfMemoryTracking` replaced with `enableWatchdogTerminationTracking`.
+  - `anrTimeoutIntervalMillis` replaced with `anrTimeoutInterval` instead.
+  - `autoSessionTrackingIntervalMillis` replaced with `autoSessionTrackingInterval` instead.
```
Comment (lizokm) [inline src/platforms/flutter/migration.mdx:18]:
```suggestion
  - `anrTimeoutIntervalMillis` replaced with `anrTimeoutInterval`.
  - `autoSessionTrackingIntervalMillis` replaced with `autoSessionTrackingInterval`.
```

File: src/platforms/dart/migration.mdx [hunk 7]
```diff
@@ -4,6 +4,43 @@ sidebar_order: 1000
...
+  - `isolate` replaced with `SentryThread#name`.
+- The following fields have been removed from the `Scope` class and replaced:
+  - `user(SentryUser? user)` replaced with `setUser(SentryUser? user)`.
+  - `attachements` replaced with `attachments`.
+- Classes or methods that used to take the bellow optional parameters, have been moved to the `SentryOptions` class:
+  - `captureFailedRequests` replaced with `SentryOptions#captureFailedRequests` instead.
+  - `sendDefaultPii` replaced with `SentryOptions#sendDefaultPii` instead.
+  - `maxRequestBodySize` replaced with `SentryOptions#maxRequestBodySize` instead.
+  - `networkTracing` replaced with `SentryOptions#tracesSampleRate` or `SentryOptions#tracesSampler` instead.
+  - `recordBreadcrumbs` replaced with `SentryOptions#recordHttpBreadcrumbs` instead.
```
Comment (lizokm) [inline src/platforms/dart/migration.mdx:28]:
```suggestion
- Classes or methods that used to take the below optional parameters, have been moved to the `SentryOptions` class and replaced:
  - `captureFailedRequests` replaced with `SentryOptions#captureFailedRequests`.
  - `sendDefaultPii` replaced with `SentryOptions#sendDefaultPii`.
  - `maxRequestBodySize` replaced with `SentryOptions#maxRequestBodySize`.
  - `networkTracing` replaced with `SentryOptions#tracesSampleRate` or `SentryOptions#tracesSampler`.
  - `recordBreadcrumbs` replaced with `SentryOptions#recordHttpBreadcrumbs`.
```

File: src/platforms/dart/migration.mdx [hunk 8]
```diff
@@ -4,6 +4,43 @@ sidebar_order: 1000
...
+-  The following`SentryMeasurementUnit`s are now strongly typed:
+  - `DurationSentryMeasurementUnit`
+  - `InformationSentryMeasurementUnit`
+  - `FractionSentryMeasurementUnit`
+  - `CustomSentryMeasurementUnit`
+  - `NoneSentryMeasurementUnit`
```
Comment (lizokm) [inline src/platforms/dart/migration.mdx:34]:
```suggestion
-  The following `SentryMeasurementUnits` are now strongly typed:
   -  `DurationSentryMeasurementUnit`
   -  `InformationSentryMeasurementUnit`
   -  `FractionSentryMeasurementUnit`
   -  `CustomSentryMeasurementUnit`
   - ` NoneSentryMeasurementUnit`
```
Want to make sure these are indented the same way the other ones are :)

**Review by lizokm — APPROVED** [review]
Looking really good! Just a few small tweaks.

**Pull request merged**